presto-main/src/main/java/io/prestosql/sql/planner/PlanOptimizers.java

```diff
@@ -526,6 +527,20 @@ public PlanOptimizers(
                         optimizerStats,
                         new PredicatePushDown(metadata, typeOperators, typeAnalyzer, false, false)));
 
+        // Perform redirection before CBO rules to ensure stats from destination connector are used
+        // Perform redirection before agg, topN, limit, sample etc. push down into table scan as the destination connector may support a different set of push downs
+        // Perform redirection after at least one PredicatePushDown and PushPredicateIntoTableScan to allow connector to use pushed down predicates in redirection decision
+        // Perform redirection after at least table scan pruning rules because redirected table might have fewer columns
+        // PushPredicateIntoTableScan must be run after any redirection
```

Comment:
Add invariant:
column pruning rules need to be run after `ApplyTableScanRedirect`